statina/API/v2/endpoints/batches.py

```diff
@@ -0,0 +1,288 @@
+from statina.crud.find.plots.coverage_plot_data import (
+    get_box_data_for_coverage_plot,
+    get_scatter_data_for_coverage_plot,
+)
+from statina.crud.find.plots.zscore_plot_data import (
+    get_tris_control_abnormal,
+    get_tris_control_normal,
+    get_tris_samples,
+)
+from statina.crud.insert import insert_batch, insert_samples
+from statina.crud.utils import zip_dir
+from statina.models.database import Batch, DataBaseSample, User
+from statina.models.server.load import BatchRequestBody
+from statina.models.server.plots.coverage import CoveragePlotSampleData
+from statina.models.server.plots.fetal_fraction import (
+    FetalFractionControlAbNormal,
+    FetalFractionSamples,
+)
+from statina.models.server.plots.fetal_fraction_sex import SexChromosomeThresholds
+from statina.models.server.sample import Sample
+from statina.parse.batch import get_batch, get_samples, validate_file_path
+
+router = APIRouter(prefix="/v2")
+
+
+@router.get("/batches")
+def batches(
+    page_size: Optional[int] = Query(5),
+    page_num: Optional[int] = Query(0),
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """List of all batches"""
+    all_batches: List[Batch] = find.batches(adapter=adapter, page_size=page_size, page_num=page_num)
+    return JSONResponse(jsonable_encoder(all_batches))
+
+
+@router.delete("/batch/{batch_id}")
+async def batch_delete(
+    batch_id: str,
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+    current_user: User = Security(get_current_active_user, scopes=["admin"]),
+):
+    await delete_batch(adapter=adapter, batch_id=batch_id)
+    return JSONResponse(content=f"Deleted batch {batch_id}", status_code=200)
+
+
+@router.post("/batch/", response_model=Batch)
+def load_batch(
+    batch_files: BatchRequestBody,
+    current_user: User = Security(get_current_active_user, scopes=["RW"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """Function to load batch data into the database with rest"""
+    nipt_results = Path(batch_files.result_file)
+    if not nipt_results.exists():
+        return JSONResponse(content="Results file missing", status_code=422)
+    samples: List[DataBaseSample] = get_samples(nipt_results)
+    batch: Batch = get_batch(nipt_results)
+    if find.batch(adapter=adapter, batch_id=batch.batch_id):
+        return JSONResponse(content="Batch already in database!", status_code=422)
+    insert_batch(adapter=adapter, batch=batch, batch_files=batch_files)
+    insert_samples(adapter=adapter, samples=samples, segmental_calls=batch_files.segmental_calls)
+    inserted_batch = find.batch(adapter=adapter, batch_id=batch.batch_id)
+    return JSONResponse(content=jsonable_encoder(inserted_batch), status_code=200)
+
+
+@router.get("/batch/{batch_id}", response_model=Batch)
+def get_batch(
+    batch_id: str,
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
```

Comment:
If this is supposed to replace the endpoint for the batch view, its not enough to return just the batch object from the database. I was intending to make responce model for this view. We can either add another endpoint called batch_view or something. But I sort of think we sohuld only create endpoints that we will use. But maybe you intendetd to separete the batch header from the batch table? in theat case we need another endpoint for the table... Lets talk!